azurelinuxagent/ga/update.py

```diff
@@ -600,11 +600,14 @@ def _process_goal_state(self, exthandlers_handler, remote_access_handler):
             protocol = exthandlers_handler.protocol
 
             # update self._goal_state
-            self._try_update_goal_state(protocol)
-
-            # Update the Guest Agent if a new version is available
-            if self._goal_state is not None:
```

Comment:
we were executing the goal state even when we fail to update it... it's a regression introduced a few releases ago